### ISSUE TITLE
Fix S2189: consider conditions that are modified asynchronously

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#2743]
```javascript
let condition = true;

async function foo() {
  /* ... */
  while (condition) { // FP: condition is modified asynchronously
    /* ...: does not modify condition */
  }
}

async function bar() {
  /* ... */
  condition = false;
}
```

Reported in https://community.sonarsource.com/t/false-positive-for-indefinite-loop-with-multiple-conditions-javascript-s2189/44970/3

**Comment by ilia-kebets-sonarsource** [comment]
we could consider ignoring the issue if the condition is a single variable.

**Comment by ilia-kebets-sonarsource** [comment]
related to https://github.com/SonarSource/SonarJS/issues/2410

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-129
